### ISSUE TITLE
[Bug]: Callout not clearing when clicking map

### ISSUE THREAD

**Issue by MariuzM** [rnmapbox/maps#3111]
### Mapbox Implementation

Mapbox

### Mapbox Version

10.12.0

### Platform

iOS, Android

### `@rnmapbox/maps` version

10.0.15

### Standalone component to reproduce

```javascript
import Mapbox from '@rnmapbox/maps';
import React from 'react';
import { StyleSheet, View } from 'react-native';

const App = () => {
  return (
    <View style={styles.container}>
      <Mapbox.MapView
        style={styles.map}
        onPress={(feature) => console.log('Coords:', feature.geometry)}
      >
        <Mapbox.PointAnnotation
          id="pointAnnotation"
          coordinate={[-0.29296964120311486, 5.9948919351710686]}
          onSelected={() => {
            // console.log('onSelected', item.id);
          }}
        >
          <View style={{ backgroundColor: 'red', width: 30, height: 30 }} />
          <Mapbox.Callout title="test" />
        </Mapbox.PointAnnotation>
      </Mapbox.MapView>
    </View>
  );
};

export default App;

const styles = StyleSheet.create({
  container: {
    height: '100%',
    width: '100%',
  },
  map: {
    flex: 1,
  },
});
```

### Observed behavior and steps to reproduce

I have noticed 2 issues but both unrelated.
1. As per title, the Callout does not disappear when clicking on the map. When i click on the red box PointAnnotation it shows Callout, but when i click on the map it does not remove it.
2. Type issue with `Mapbox.MapView` onPress, coordinates is available but not in TypeScript
<img width="719" alt="image" src="https://github.com/rnmapbox/maps/assets/36937039/03df6bc8-3e2a-4c68-ba72-5802680e06fd">

### Expected behavior

1. When clicking outside the Callout should disappear
2. Should be a type of  `coordinates`

**Comment by MariuzM** [comment]
Another issue i have is when i want to have a button on Callout it does not work, i mean i dont get the button click event. Feels like map is just overlaying that button with its own touch layer.

**Comment by phucprime** [comment]
I'm not facing this issue on Android.
On iOS I've fixed by call removeAll() callout on MapView press

`RNMBXMapView.swift`

```
@objc
  func doHandleTap(_ sender: UITapGestureRecognizer) {
    self.calloutAnnotationManager.annotations.removeAll() // Add this line
    let tapPoint = sender.location(in: self)
    ...
```

**Comment by mfazekas** [comment]
Should be fixed by: #3258
